File: LeetCode_3147/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maximumEnergy(vector<int>& energy, int k) {
        int n = energy.size();
        int answer = INT_MIN;

        for (int i = n - 1; i >= n - k; i--) {
            int sum = 0;
            for (int j = i; j >= 0; j -= k) {
                sum += energy[j];
                answer = max(answer, sum);
            }
        }
        return answer;
    }
};

File: LeetCode_3147/Main.py
class Solution(object):
    def maximumEnergy(self, energy, k):
        """
        :type energy: List[int]
        :type k: int
        :rtype: int
        """
        n = len(energy)
        answer = energy[n - 1]

        for i in range(n - 1, n - k - 1, -1):
            sum = 0
            j = i
            while j >= 0:
                sum += energy[j]
                answer = max(answer, sum)
                j -= k
        return answer